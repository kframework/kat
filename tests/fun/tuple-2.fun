// testing functions taking tuple arguments

datatype ('a,'b,'c) triple = Triple 'a 'b 'c

let f (Triple x y z) = x + y + z * y + z - x + z * z - y
 in f (Triple 1 2 3)

// 1 + 2 + (3 * 2) + 3 - 1 + (3 * 3) - 2
// 1 + 2 + 6       + 3 - 1 + 9       - 2
// 9               + 11              - 2
// 18
