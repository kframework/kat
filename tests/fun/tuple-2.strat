a0e64a3058ae311eb0ad475cb50f3fee orig
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx2 exec
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx3 compile
