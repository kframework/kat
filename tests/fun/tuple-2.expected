
Strategy: orig
================================================================================
<FUN>
  <k>
    18
  </k>
  <env>
    .Map
  </env>
  <envs>
    .List
  </envs>
</FUN>

Strategy: exec
================================================================================
<kat-FUN>
  <s>
    #STUCK ( )
  </s>
  <kat>
    <analysis>
      .Analysis
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <FUN>
      <k>
        18
      </k>
      <env>
        .Map
      </env>
      <envs>
        .List
      </envs>
    </FUN>
  </harness>
</kat-FUN>

Strategy: compile
================================================================================
<kat-FUN>
  <s>
    #STUCK ( )
  </s>
  <kat>
    <analysis>
      .Rules , < <FUN>
        <k>
          let ( f ( Triple x y z ) = x + y + z * y + z - x + z * z - y ) and .Bindings in f ( Triple 1 2 3 )
        </k>
        <env>
          .Map
        </env>
        <envs>
          .List
        </envs>
      </FUN> --> <FUN>
        <k>
          18
        </k>
        <env>
          .Map
        </env>
        <envs>
          .List
        </envs>
      </FUN> requires #True >
    </analysis>
    <states>
      .States
    </states>
  </kat>
  <harness>
    <FUN>
      <k>
        18
      </k>
      <env>
        .Map
      </env>
      <envs>
        .List
      </envs>
    </FUN>
  </harness>
</kat-FUN>
